### DISCUSSION TITLE
The search result of E-Hentai Plugin is not correct.

### DISCUSSION THREAD

**Discussion by PasserDreamer** [Difegue/LANraragi#578] [General]
**LRR Version and OS**
Docker Version 0.8.3

**Bug Details**
The search result in sad panda messed up when mixing English and Japanese as keywords.
(e.g. The title with "JK" all got the same result.)
After doing some research, I found that sad panda search both title and tag by default.
To resolve this issue, use advanced search option.
(https://sad🐼/?f_cats=0&f_search={Keyword Here}&advsearch=1&f_sname=on)

**Comment by Difegue** [comment]
Ad you can see [here](https://github.com/Difegue/LANraragi/blob/2465b56862a6e807529558c791ad86c249d5b4a5/lib/LANraragi/Plugin/Metadata/EHentai.pm#L136), the metadata plugin does use advanced search by default.  

If you enable "debug mode" in server settings, the plugin logs will show the full search URLs used; I recommend you take a look at those and see how it differs from what you're expecting. We might be able to make some improvements then 👍

**Comment by PasserDreamer** [comment]
https://github.com/Difegue/LANraragi/blob/2465b56862a6e807529558c791ad86c249d5b4a5/lib/LANraragi/Plugin/Metadata/EHentai.pm#L136

Remove `&f_stags=on` will be ok since tag search is not required.

`[2021-12-24 08:21:22] [E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_stags=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_search=%22%5B%E3%81%84%E3%81%A1%E3%81%94%E3%82%AF%E3%83%AC%E3%83%BC%E3%83%97%E5%A4%A7%E7%9B%9B%E7%B5%84%20%28%E6%A8%AA%E5%8D%81%E8%BC%94%29%5D%20%E6%8F%B4%E4%BA%A4JK%E3%81%AB%E7%B5%B6%E5%AF%BE%E4%B8%AD%E5%87%BA%E3%81%97%E3%81%99%E3%82%8B%E3%83%9E%E3%83%B3%EF%BC%81%20%E3%80%8C%E3%81%8A%E3%81%A3%E3%81%A8%E3%82%8A%E3%81%8A%E3%83%90%E3%82%AB%E7%B3%BB%E3%81%BD%E3%81%A1%E3%82%83%E5%A8%98%E7%B7%A8%E3%80%8D%22&f_sh=on+language%3Ajapanese (archive title)
[2021-12-24 08:21:24] [E-Hentai] [debug] EH API Tokens are 1826976 / 4ca07a70c7
[2021-12-24 08:21:25] [E-Hentai] [debug] E-H API returned this JSON: {"gmetadata":[{"gid":1826976,"token":"4ca07a70c7","archiver_key":"455648--4f54b2c884a897d43771c8a80f5b0cae55efa7f5","title":"ARTIST JK","title_jpn":"","category":"Western","thumb":"https:\/\/ehgt.org\/09\/c3\/09c3b195bd15fcfd9069be2c4ade315e89c8c5b8-574607-900-841-jpg_l.jpg","uploader":"ranma-chan","posted":"1633193205","filecount":"816","filesize":268997262,"expunged":false,"rating":"2.88","torrentcount":"0","torrents":[],"tags":["male:shotacon","female:lolicon","mixed:incest","other:western imageset"]}]}
[2021-12-24 08:21:25] [E-Hentai] [info] Sending the following tags to LRR: male:shotacon, female:lolicon, mixed:incest, other:western imageset, category:western`

**Comment by polak14** [comment]
tag search is required when searching for language.

Anyway since this is about the EH plugin, a suggestion i would like to make is stripping gallery IDs from title. Its a problem when you download galleries via h@h.
for example: "[Hardcore Zayaku Souten (Hirayan)] 45XY45 (Girls' Frontline) [English] [MegaFagget] [Digital] [1543221]" will return no results but "[Hardcore Zayaku Souten (Hirayan)] 45XY45 (Girls' Frontline) [English] [MegaFagget] [Digital]" will. 

`$title =~ s/ \[[0-9]+\]//g;` i just copied this part from the fakku plugin and adjusted it for my needs, it works fine for me. 

Also there appears to be no delay between thumbnail and title search and I get banned fairly often due to it.

**Comment by Difegue** [comment]
It's annoying that h@h only gives the gallery ID, since if we had the gallery token alongside it we could just hit the API directly and skip the thumbnail/title search.  
I don't think there's a way to get the token, but someone might know more about this than I do.  

As for delays, there is normally one, but it depends on whether you receive the excessive request warning first:  
https://github.com/Difegue/LANraragi/blob/959b79455b0a736e18c1b7273506690117f9f41b/lib/LANraragi/Plugin/Metadata/EHentai.pm#L215

**Comment by FeudalNoodle** [comment]
Regarding the H@H issue, I'm currently looking into bastardizing one of the JSON plugins.
Each gallery downloaded via H@H comes with a galleryinfo.txt containing a snapshot of all relevant metadata at the time of download.

The .txt files look like this:
> Title:      xxxxxxxxxx
> Upload Time: 2020-11-11 00:00
> Uploaded By: Katlan
> Downloaded:  2021-12-26 22:09
> Tags:        language:english, language:translated, artist:yyyy, male:dark skin, female:fox girl, [...]
> 
> Uploader's Comments:
> 
> [...]
> 
> Downloaded from E-Hentai Galleries by the Hentai@Home Downloader <3
> 

I'll haven't really dabbled with REGEX or perl before so it may take a solid amount of fiddling.

**Comment by Difegue** [comment]
Not really; Looking at the format it's likely you can just hack on the HDoujin plugin which has a similar format.  

Simply editing this line: 
https://github.com/Difegue/LANraragi/blob/959b79455b0a736e18c1b7273506690117f9f41b/lib/LANraragi/Plugin/Metadata/Hdoujin.pm#L75  
with `galleryinfo.txt` and this one:
https://github.com/Difegue/LANraragi/blob/959b79455b0a736e18c1b7273506690117f9f41b/lib/LANraragi/Plugin/Metadata/Hdoujin.pm#L88 
to use `Tags` instead of `TAGS` should be enough to get this working in a basic way.   
You can then extend to add support for `Title` and `Upload Time`, likely.  

Bundling dedicated plugins for all possible variants of .txt/.json files isn't viable long term as it seems a different one pops up every 6 months. This is a case where having a real plugin repository would certainly help..

**Comment by Difegue** [comment]
This issue is turning into a dumping grounds for EH issues so I'm converting it to a discussion -- as far as the original issue goes, I've added an adjustment to disable `f_stags` unless default language is toggled.

**Comment by pandafeliz** [comment]
I tried using this version plugin and had to revert to a previous version since I was getting a lot of "not found" hits on files that would usually had no problem fetching the metadata.

**Reply by Difegue** [reply]
Could you turn on debug mode and provide URLs from the previous version and the current one where the problem arises? It'd help a lot in figuring out how the whole shebang works.  

That or I can just revert the changes I guess tbh

**Reply by pandafeliz** [reply]
yeah, I can do that, the main culprit with this version seems when the filename differs from the gallery name.

**Comment by pandafeliz** [comment]
Settings I use for the EH plugin
![chrome_gTC5nebzzL](https://user-images.githubusercontent.com/72732463/147995044-97bf51cc-3bab-4757-9e1f-9eb653a46cc4.png)

Here is the log using the plugin version from this commit 959b79455b0a736e18c1b7273506690117f9f41b

`[Batch Tagging] [info] Client connected to Batch Tagging service
[2022-01-04 00:37:17.54752] [36] [trace] [qzFyy5ZRZwqT] 101 Switching Protocols (0.006735s, 148.478/s)
[Mojolicious] [qzFyy5ZRZwqT]
[Batch Tagging] [debug] Processing b31ec674692c4555991c0ce060311d8573e66f7a
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for b31ec674692c4555991c0ce060311d8573e66f7a from 001.jpg
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/b31ec674692c4555991c0ce060311d8573e66f7a/001.jpg
[Archive] [debug] Found file 001.jpg in archive /mnt/g/test/(SC38) [Kohakutei (Sakai Hamachi)] Ai ga Horo Horo 2 (Spice and Wolf) [English] [Trinity Translations].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_stags=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=f034444d108310a0b8f810fd556544beabbce184&fs_covers=1&fs_similar=1&f_search=language%3Aenglish (archive thumbnail hash)
[E-Hentai] [debug] EH API Tokens are 378582 / 244b06a7d9
[E-Hentai] [debug] E-H API returned this JSON: {"gmetadata":[{"gid":378582,"token":"244b06a7d9","archiver_key":"455904--36ae152aeec6479c1a61778423f2699ba8333215","title":"(SC38) [Kohakutei (Sakai Hamachi)] Ai ga Horohoro (Spice and Wolf) [English] [Trinity Translations]","title_jpn":"(\u30b5\u30f3\u30af\u30ea38) [\u7425\u73c0\u4ead (\u583a\u306f\u307e\u3061)] \u611b\u304c\u30db\u30ed\u30db\u30ed (\u72fc\u3068\u9999\u8f9b\u6599) [\u82f1\u8a33]","category":"Doujinshi","thumb":"https:\/\/ehgt.org\/f0\/34\/f034444d108310a0b8f810fd556544beabbce184-139038-989-1400-jpg_l.jpg","uploader":"N04h","posted":"1308297369","filecount":"19","filesize":3546215,"expunged":false,"rating":"4.78","torrentcount":"1","torrents":[{"hash":"2051d0527bc87c7fa3fec4236d45e40f07ddb282","added":"1308298286","name":"[Kohakutei (Sakai Hamachi)] Ai ga Horo Horo 2 (Spice and Wolf)(Eng)(Trinity Translations).zip","tsize":"1493","fsize":"3548601"}],"tags":["language:english","language:translated","parody:spice and wolf","character:holo","character:kraft lawrence","group:kohakutei","artist:sakai hamachi","male:sole male","female:kemonomimi","female:small breasts","female:sole female","female:tail","female:wolf girl"]}]}
[E-Hentai] [info] Sending the following tags to LRR: language:english, language:translated, parody:spice and wolf, character:holo, character:kraft lawrence, group:kohakutei, artist:sakai hamachi, male:sole male, female:kemonomimi, female:small breasts, female:sole female, female:tail, female:wolf girl, category:doujinshi
[Plugin System] [info] Applying tag rules...
[Batch Tagging] [debug] Processing ab47e757db528cdd29c2bffd51de8a3605d091e0
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for ab47e757db528cdd29c2bffd51de8a3605d091e0 from 0001.jpg
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/ab47e757db528cdd29c2bffd51de8a3605d091e0/0001.jpg
[Archive] [debug] Found file 0001.jpg in archive /mnt/g/test/[Massaratou] Honenuki Sakusen!   Boning Strategy! [English] {doujin-moe.us} [Digital].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_stags=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=a4486e4f17d826b04dff57de6b40f9b4a9548bdc&fs_covers=1&fs_similar=1&f_search=language%3Aenglish (archive thumbnail hash)
[E-Hentai] [debug] EH API Tokens are 701954 / 1d857a06c0
[E-Hentai] [debug] E-H API returned this JSON: {"gmetadata":[{"gid":701954,"token":"1d857a06c0","archiver_key":"455904--0e067cccc51f0980e760e036ec86765aa8d3d34b","title":"[Massaratou (Motomushi)] Honenuki Sakusen! | Boning Strategy! [English] {doujin-moe.us}","title_jpn":"[\u307e\u3063\u3055\u3089\u5cf6 (\u5143\u866b)] \u9aa8\u629c\u304d\u4f5c\u6226! [\u82f1\u8a33]","category":"Doujinshi","thumb":"https:\/\/ehgt.org\/a4\/48\/a4486e4f17d826b04dff57de6b40f9b4a9548bdc-1567091-1256-1754-jpg_l.jpg","uploader":"Drksrpnt","posted":"1400398961","filecount":"24","filesize":16050540,"expunged":false,"rating":"4.76","torrentcount":"1","torrents":[{"hash":"dd2fa220fcbc651a737ea5f1ae3fa4fd5e4fcf7a","added":"1497874336","name":"[Massaratou] Honenuki Sakusen!   Boning Strategy! [English] {doujin-moe.us} [Digital].zip","tsize":"4969","fsize":"14976998"}],"tags":["language:english","language:translated","parody:original","group:massaratou","artist:motomushi","female:anal","female:big breasts","female:bukkake","female:double penetration","female:sex toys","female:stockings","female:x-ray","mixed:group","mixed:mmf threesome"]}]}
[E-Hentai] [info] Sending the following tags to LRR: language:english, language:translated, parody:original, group:massaratou, artist:motomushi, female:anal, female:big breasts, female:bukkake, female:double penetration, female:sex toys, female:stockings, female:x-ray, mixed:group, mixed:mmf threesome, category:doujinshi
[Plugin System] [info] Applying tag rules...
[Batch Tagging] [debug] Processing f6e82245c98386096aae0c5cc5e7fefdb84c0e1d
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for f6e82245c98386096aae0c5cc5e7fefdb84c0e1d from SaHa_Mahou_Shoujo_1.02.0_01.jpg
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/f6e82245c98386096aae0c5cc5e7fefdb84c0e1d/SaHa_Mahou_Shoujo_1.02.0_01.jpg
[Archive] [debug] Found file SaHa_Mahou_Shoujo_1.02.0_01.jpg in archive /mnt/g/test/(C70) [Zettai Shoujo (RAITA)] Mahou shoujo soushuuhen [English] [SaHa].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_stags=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=750e5faaeca0ebfc374387f8bf91e6341b54787c&fs_covers=1&fs_similar=1&f_search=language%3Aenglish (archive thumbnail hash)
[E-Hentai] [debug] EH API Tokens are 342618 / 03ad7ce780
[E-Hentai] [debug] E-H API returned this JSON: {"gmetadata":[{"gid":342618,"token":"03ad7ce780","archiver_key":"455904--e307e5af25f3aed443204039e383c3d336d25b34","title":"(C70) [Zettai Shoujo (Raita)] Mahou Shoujo Soushuuhen (Zettai Junpaku Mahou Shoujo) [English] [SaHa]","title_jpn":"(C70) [\u7d76\u5bfe\u5c11\u5973 (RAITA)] \u9b54\u6cd5\u5c11\u5973\u7dcf\u96c6\u7de8 (\u7d76\u5bfe\u7d14\u767d\u30fb\u9b54\u6cd5\u5c11\u5973) [\u82f1\u8a33]","category":"Doujinshi","thumb":"https:\/\/ehgt.org\/75\/0e\/750e5faaeca0ebfc374387f8bf91e6341b54787c-387722-1050-1500-jpg_l.jpg","uploader":"Eutopia","posted":"1299259816","filecount":"58","filesize":24587054,"expunged":false,"rating":"4.70","torrentcount":"1","torrents":[{"hash":"4f16217d1b5e5557bb76ee53b239c44062c5a383","added":"1299266312","name":"[Raita] Mahou Shoujo 1.0&amp;2.0 [English] [SaHa].zip","tsize":"7870","fsize":"24597052"}],"tags":["language:english","language:translated","parody:zettai junpaku mahou shoujo","character:misa suzuhara","character:sasaki kotone","group:zettai shoujo","artist:raita","male:sole male","female:defloration","female:impregnation","female:lolicon","female:nakadashi","female:sweating","female:wings","female:x-ray","other:multi-work series"]}]}
[E-Hentai] [info] Sending the following tags to LRR: language:english, language:translated, parody:zettai junpaku mahou shoujo, character:misa suzuhara, character:sasaki kotone, group:zettai shoujo, artist:raita, male:sole male, female:defloration, female:impregnation, female:lolicon, female:nakadashi, female:sweating, female:wings, female:x-ray, other:multi-work series, category:doujinshi
[Plugin System] [info] Applying tag rules...
[Batch Tagging] [debug] Processing 5c90fdf98e092661aba01509ebe00ed3296c5cab
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for 5c90fdf98e092661aba01509ebe00ed3296c5cab from 001.png
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/5c90fdf98e092661aba01509ebe00ed3296c5cab/001.png
[Archive] [debug] Found file 001.png in archive /mnt/g/test/(COMITIA116) [ZOAL (LENA[A-7])] aru ballet kyoushitsu no super YSJ tokkun to yara   It's the ballet class's super YSJ training [English] [#Based Anons].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_stags=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=e8e845389fe2f58ed436133ae21208002f392570&fs_covers=1&fs_similar=1&f_search=language%3Aenglish (archive thumbnail hash)
[E-Hentai] [debug] EH API Tokens are 1021671 / 004bd8f658
[E-Hentai] [debug] E-H API returned this JSON: {"gmetadata":[{"gid":1021671,"token":"004bd8f658","archiver_key":"455904--f568483dc520fcf9b1f229964973482ce5cd8693","title":"(COMITIA116) [ZOAL (LENA[A-7])] aru ballet kyoushitsu no super YSJ tokkun to yara | It&#039;s the ballet class&#039;s super YSJ training [English]","title_jpn":"(\u30b3\u30df\u30c6\u30a3\u30a2116) [ZOAL (LENA[A-7])] \u3042\u308b\u30d0\u30ec\u30a8\u6559\u5ba4\u306e\u30b9\u30fc\u30d1\u30fc\uff39\uff33\uff2a\u7279\u8a13\u3068\u3084\u3089 [\u82f1\u8a33]","category":"Doujinshi","thumb":"https:\/\/ehgt.org\/e8\/e8\/e8e845389fe2f58ed436133ae21208002f392570-1033040-2082-3000-png_l.jpg","uploader":"1\/4-Based Anon","posted":"1485189167","filecount":"13","filesize":13293980,"expunged":false,"rating":"4.63","torrentcount":"1","torrents":[{"hash":"47bd365f55bdb91377c55264476b57516c4bc841","added":"1485194740","name":"(COMITIA116) [ZOAL (LENA[A-7])] aru ballet kyoushitsu no super YSJ tokkun to yara\uff5cIt&#039;s the ballet class&#039;s super YSJ training [English] [#Based Anons]","tsize":"4967","fsize":"13293980"}],"tags":["language:english","language:translated","parody:original","group:zoal","artist:lena a-7","male:coach","male:sole male","female:hair buns","female:leotard","female:pantyhose","female:sole female","female:sweating"]}]}
[E-Hentai] [info] Sending the following tags to LRR: language:english, language:translated, parody:original, group:zoal, artist:lena a-7, male:coach, male:sole male, female:hair buns, female:leotard, female:pantyhose, female:sole female, female:sweating, category:doujinshi
[Plugin System] [info] Applying tag rules...
[Batch Tagging] [info] Client disconnected, halting remaining operations
[2022-01-04 00:37:47.74220] [37] [trace] [_O_TMNypnSbj] DELETE "/api/search/cache"
[Mojolicious] [_O_TMNypnSbj]
[2022-01-04 00:37:47.74342] [37] [trace] [_O_TMNypnSbj] Routing to controller "LANraragi::Controller::Login" and action "logged_in_api"
[Mojolicious] [_O_TMNypnSbj]
[2022-01-04 00:37:47.74540] [37] [trace] [_O_TMNypnSbj] Routing to controller "LANraragi::Controller::Api::Search" and action "clear_cache"
[Mojolicious] [_O_TMNypnSbj]
[2022-01-04 00:37:47.76996] [37] [trace] [_O_TMNypnSbj] 200 OK (0.027733s, 36.058/s)
[Mojolicious] [_O_TMNypnSbj]
Mojo::Reactor::Poll: I/O watcher failed: Can't call method "emit" on an undefined value at /usr/local/share/perl5/site_perl/Mojo/Redis.pm line 60.
Mojo::Reactor::Poll: I/O watcher failed: Can't call method "emit" on an undefined value at /usr/local/share/perl5/site_perl/Mojo/Redis.pm line 60.
[2022-01-04 00:37:50.83658] [32] [debug] Process 44 is performing job "3" with task "warm_cache"
[Mojolicious] Process 44 is performing job "3" with task "warm_cache"
[Minion] [info] Warming up search cache...
[Search Engine] [debug] Search request: --title-asc-0-0
[Search Engine] [debug] No cache available, doing a full DB parse.
[2022-01-04 00:37:50.87314] [32] [debug] Process 46 is performing job "4" with task "build_stat_hashes"
[Mojolicious] Process 46 is performing job "4" with task "build_stat_hashes"
[Tag Stats] [debug] Building stat indexes...
[Tag Stats] [debug] Adding exhentai.org/g/1021671/004bd8f658 as an URL for 5c90fdf98e092661aba01509ebe00ed3296c5cab
[Tag Stats] [debug] Adding exhentai.org/g/342618/03ad7ce780 as an URL for f6e82245c98386096aae0c5cc5e7fefdb84c0e1d
[Tag Stats] [debug] Adding exhentai.org/g/701954/1d857a06c0 as an URL for ab47e757db528cdd29c2bffd51de8a3605d091e0
[Tag Stats] [debug] Adding exhentai.org/g/378582/244b06a7d9 as an URL for b31ec674692c4555991c0ce060311d8573e66f7a
[Tag Stats] [debug] Done!
[2022-01-04 00:37:51.72140] [20] [info] Creating process id file "/home/koyomi/lanraragi/public/temp/server.pid"
[Mojolicious] Creating process id file "/home/koyomi/lanraragi/public/temp/server.pid"
[Minion] [info] Done!`

and here is the log from using the plugin version of this commit d2719567320481952c07af4ee296ef9a14b8e171

`[Batch Tagging] [info] Client connected to Batch Tagging service
[2022-01-04 00:31:46.64562] [31] [trace] [UthlXD4PYGc3] 101 Switching Protocols (0.006089s, 164.230/s)
[Mojolicious] [UthlXD4PYGc3]
[Batch Tagging] [debug] Processing f6e82245c98386096aae0c5cc5e7fefdb84c0e1d
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for f6e82245c98386096aae0c5cc5e7fefdb84c0e1d from SaHa_Mahou_Shoujo_1.02.0_01.jpg
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/f6e82245c98386096aae0c5cc5e7fefdb84c0e1d/SaHa_Mahou_Shoujo_1.02.0_01.jpg
[Archive] [debug] Found file SaHa_Mahou_Shoujo_1.02.0_01.jpg in archive /mnt/g/test/(C70) [Zettai Shoujo (RAITA)] Mahou shoujo soushuuhen [English] [SaHa].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=750e5faaeca0ebfc374387f8bf91e6341b54787c&fs_covers=1&fs_similar=1&f_search=language%3Aenglish%26f_stags%3Don (archive thumbnail hash)
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_search=%22%28C70%29%20%5BZettai%20Shoujo%20%28RAITA%29%5D%20Mahou%20shoujo%20soushuuhen%20%5BEnglish%5D%20%5BSaHa%5D%22+language%3Aenglish&f_stags=on (archive title)
[E-Hentai] [info] No matching EH Gallery Found!
[Batch Tagging] [debug] Processing b31ec674692c4555991c0ce060311d8573e66f7a
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for b31ec674692c4555991c0ce060311d8573e66f7a from 001.jpg
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/b31ec674692c4555991c0ce060311d8573e66f7a/001.jpg
[Archive] [debug] Found file 001.jpg in archive /mnt/g/test/(SC38) [Kohakutei (Sakai Hamachi)] Ai ga Horo Horo 2 (Spice and Wolf) [English] [Trinity Translations].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=f034444d108310a0b8f810fd556544beabbce184&fs_covers=1&fs_similar=1&f_search=language%3Aenglish%26f_stags%3Don (archive thumbnail hash)
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_search=%22%28SC38%29%20%5BKohakutei%20%28Sakai%20Hamachi%29%5D%20Ai%20ga%20Horo%20Horo%202%20%28Spice%20and%20Wolf%29%20%5BEnglish%5D%20%5BTrinity%20Translations%5D%22+language%3Aenglish&f_stags=on (archive title)
[E-Hentai] [info] No matching EH Gallery Found!
[Batch Tagging] [debug] Processing ab47e757db528cdd29c2bffd51de8a3605d091e0
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for ab47e757db528cdd29c2bffd51de8a3605d091e0 from 0001.jpg
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/ab47e757db528cdd29c2bffd51de8a3605d091e0/0001.jpg
[Archive] [debug] Found file 0001.jpg in archive /mnt/g/test/[Massaratou] Honenuki Sakusen!   Boning Strategy! [English] {doujin-moe.us} [Digital].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=a4486e4f17d826b04dff57de6b40f9b4a9548bdc&fs_covers=1&fs_similar=1&f_search=language%3Aenglish%26f_stags%3Don (archive thumbnail hash)
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_search=%22%5BMassaratou%5D%20Honenuki%20Sakusen%21%20%20%20Boning%20Strategy%21%20%5BEnglish%5D%20%7Bdoujin-moe.us%7D%20%5BDigital%5D%22+language%3Aenglish&f_stags=on (archive title)
[E-Hentai] [info] No matching EH Gallery Found!
[Batch Tagging] [debug] Processing 5c90fdf98e092661aba01509ebe00ed3296c5cab
[Batch Tagging] [debug] No user overrides given.
[Plugin System] [info] Thumbnail hash invalid, regenerating.
[Archive] [debug] Extracting thumbnail for 5c90fdf98e092661aba01509ebe00ed3296c5cab from 001.png
[Archive] [debug] Output for single file extraction: /home/koyomi/lanraragi/public/temp/thumb/5c90fdf98e092661aba01509ebe00ed3296c5cab/001.png
[Archive] [debug] Found file 001.png in archive /mnt/g/test/(COMITIA116) [ZOAL (LENA[A-7])] aru ballet kyoushitsu no super YSJ tokkun to yara   It's the ballet class's super YSJ training [English] [#Based Anons].zip
[Plugin System] [debug] Calling matching login plugin ehlogin.
[E-Hentai Login] [info] Cookies provided ()!
[E-Hentai] [info] Reverse Image Search Enabled, trying now.
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_shash=e8e845389fe2f58ed436133ae21208002f392570&fs_covers=1&fs_similar=1&f_search=language%3Aenglish%26f_stags%3Don (archive thumbnail hash)
[E-Hentai] [debug] Using URL https://exhentai.org?advsearch=1&f_sname=on&f_sdt2=on&f_spf=&f_spt=&f_sfu=on&f_sft=on&f_sfl=on&f_search=%22%28COMITIA116%29%20%5BZOAL%20%28LENA%5BA-7%5D%29%5D%20aru%20ballet%20kyoushitsu%20no%20super%20YSJ%20tokkun%20to%20yara%20%20%20It%27s%20the%20ballet%20class%27s%20super%20YSJ%20training%20%5BEnglish%5D%20%5B%23Based%20Anons%5D%22+language%3Aenglish&f_stags=on (archive title)
[E-Hentai] [info] No matching EH Gallery Found!
[Batch Tagging] [info] Client disconnected, halting remaining operations
[2022-01-04 00:32:08.59738] [32] [trace] [vXrAUBnenTZt] DELETE "/api/search/cache"
[Mojolicious] [vXrAUBnenTZt]
[2022-01-04 00:32:08.59856] [32] [trace] [vXrAUBnenTZt] Routing to controller "LANraragi::Controller::Login" and action "logged_in_api"
[Mojolicious] [vXrAUBnenTZt]
[2022-01-04 00:32:08.60052] [32] [trace] [vXrAUBnenTZt] Routing to controller "LANraragi::Controller::Api::Search" and action "clear_cache"
[Mojolicious] [vXrAUBnenTZt]
[2022-01-04 00:32:08.61910] [32] [trace] [vXrAUBnenTZt] 200 OK (0.021707s, 46.068/s)
[Mojolicious] [vXrAUBnenTZt]
Mojo::Reactor::Poll: I/O watcher failed: Can't call method "emit" on an undefined value at /usr/local/share/perl5/site_perl/Mojo/Redis.pm line 60.
Mojo::Reactor::Poll: I/O watcher failed: Can't call method "emit" on an undefined value at /usr/local/share/perl5/site_perl/Mojo/Redis.pm line 60.
[2022-01-04 00:32:08.79686] [27] [debug] Process 42 is performing job "3" with task "warm_cache"
[Mojolicious] Process 42 is performing job "3" with task "warm_cache"
[Minion] [info] Warming up search cache...
[Search Engine] [debug] Search request: --title-asc-0-0
[Search Engine] [debug] No cache available, doing a full DB parse.
[2022-01-04 00:32:08.83418] [27] [debug] Process 44 is performing job "4" with task "build_stat_hashes"
[Mojolicious] Process 44 is performing job "4" with task "build_stat_hashes"
[Tag Stats] [debug] Building stat indexes...
[Tag Stats] [debug] Done!
[2022-01-04 00:32:09.72142] [15] [info] Creating process id file "/home/koyomi/lanraragi/public/temp/server.pid"
[Mojolicious] Creating process id file "/home/koyomi/lanraragi/public/temp/server.pid"
[Minion] [info] Done!`

Let me know if I can provide any more info.

**Reply by Difegue** [reply]
Awesome, this should be enough. Thanks a lot! I'll take a detailed look later.

**Reply by Difegue** [reply]
@pandafeliz Seems like the thumbnail search URLs had an encoding issue so the searches weren't going through properly - That should be fixed now.  

I've also taken the opportunity to fix some issues with plugin exec order that should make the fallback text searches work better as well. (There were some regressions caused by moving regexparse to a plugin instead of a built-in server functionality)

**Reply by pandafeliz** [reply]
Thanks a lot, I still have around 9k untagged files so I give it a try later.
